lib/rules/no-new-buffer.js

```diff
@@ -0,0 +1,21 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+module.exports = {
+  create(context) {
+    return {
+      NewExpression(node) {
+        const callee = node.callee
+        if (callee && callee.name === 'Buffer') {
+          context.report({
+            node: node,
+            message: 'new Buffer() is deprecated'
```

Comment:
@divyabiyani we also need to create/include a test file. example: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/test/no-new-buffer.js